File: AtlasWeb/Core/Modules/Tibbi_Surec/Ameliyat_Modulu/SurgeryAppointmentFormViewModel.cs
//$8FFCDC5B
using System;
using System.Linq;
using Core.Models;
using Infrastructure.Filters;
using Infrastructure.Models;
using TTInstanceManagement;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;

namespace Core.Controllers
{
    public partial class SurgeryAppointmentServiceController
    {
        partial void PreScript_SurgeryAppointmentForm(SurgeryAppointmentFormViewModel viewModel, SurgeryAppointment surgeryAppointment, TTObjectContext objectContext)
        {
            Guid? selectedEpisodeActionObjectID = viewModel.GetSelectedEpisodeActionID();
            Guid? selectedPatientObjectID = viewModel.GetSelectedPatientID();
            if (selectedEpisodeActionObjectID.HasValue) //Randevu Action üzerinden açýldýðýnda
            {
                EpisodeAction episodeAction = objectContext.GetObject<EpisodeAction>(selectedEpisodeActionObjectID.Value);
                viewModel.PatientObject = episodeAction.Episode.Patient;
            }
            else if (selectedPatientObjectID.HasValue) //Randevu Action üzerinden açýlmadýðýnda
            {
                viewModel.PatientObject = objectContext.GetObject<Patient>(selectedPatientObjectID.Value);
            }

            viewModel._SurgeryAppointment.CreatedBy = Common.CurrentResource;
            viewModel._SurgeryAppointment.CreatedDate = Common.RecTime();
            viewModel._SurgeryAppointment.Patient = viewModel.PatientObject;
            if (surgeryAppointment.IsNeedAnesthesia == null)
                surgeryAppointment.IsNeedAnesthesia = true;
            bool isHeadDoctorApproveActive = TTObjectClasses.SystemParameter.GetParameterValue("AMELIYATRANDEVUBASHEKIMONAYAKTIF", "FALSE") == "FALSE" ? false : true;


            if (Common.CurrentUser.HasRole(TTRoleNames.Tabip))
            {
                if (isHeadDoctorApproveActive == true)
                    surgeryAppointment.CurrentStateDefID = SurgeryAppointment.States.HeadDoctorApproval;
                else
                    surgeryAppointment.CurrentStateDefID = SurgeryAppointment.States.Approved;
            }
        }

        partial void PostScript_SurgeryAppointmentForm(SurgeryAppointmentFormViewModel viewModel, SurgeryAppointment surgeryAppointment, TTDefinitionManagement.TTObjectStateTransitionDef transDef, TTObjectContext objectContext)
        {
            foreach (SurgeryAppointmentProc proc in surgeryAppointment.SurgeryAppointmentProcedures)
            {
                if (proc.ProcedureDefinition.ProcedureMaterialMatches.Count > 0)
                {
                    //Hizmet malzeme eþleþmesinden gelen malzemeleri randevuya eklemek için
                    foreach (ProcedureMaterialMatch match in proc.ProcedureDefinition.ProcedureMaterialMatches)
                    {
                        SurgeryAppointmentMaterial surgeryMaterial = new SurgeryAppointmentMaterial(objectContext);
                        surgeryMaterial.Procedure = match.Procedure;
                        surgeryMaterial.Material = match.Material;
                        surgeryMaterial.Store = match.Store;
                        surgeryMaterial.Amount = match.Amount;
                        surgeryMaterial.SurgeryAppointment = surgeryAppointment;
                    }
                    objectContext.Save();
                }
            }

        }
    }
}

namespace Core.Models
{
    public partial class SurgeryAppointmentFormViewModel : BaseViewModel
    {
        public Patient PatientObject { get; set; }
    }
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/Ameliyat_Modulu/SurgeryAppointmentFormViewModel.ts
//$1E6CECB1
import { SurgeryAppointment, Patient } from "NebulaClient/Model/AtlasClientModel";
import { SurgeryAppointmentProc } from "NebulaClient/Model/AtlasClientModel";
import { ProcedureDefinition } from "NebulaClient/Model/AtlasClientModel";
import { ResSection } from "NebulaClient/Model/AtlasClientModel";
import { ResUser } from "NebulaClient/Model/AtlasClientModel";
import { ResSurgeryRoom } from "NebulaClient/Model/AtlasClientModel";
import { ResSurgeryDesk } from "NebulaClient/Model/AtlasClientModel";
import { BaseViewModel } from "app/NebulaClient/Model/BaseViewModel";
import { Type } from "app/NebulaClient/ClassTransformer";

export class SurgeryAppointmentFormViewModel extends BaseViewModel {
    @Type(()=>SurgeryAppointment)
    public _SurgeryAppointment: SurgeryAppointment = new SurgeryAppointment();
    @Type(()=>Patient)
    public PatientObject: Patient = new Patient();
    @Type(()=>SurgeryAppointmentProc)
    public SurgeryAppointmentProceduresGridList: Array<SurgeryAppointmentProc> = new Array<SurgeryAppointmentProc>();
    @Type(()=>ProcedureDefinition)
    public ProcedureDefinitions: Array<ProcedureDefinition> = new Array<ProcedureDefinition>();
    @Type(()=>ResSection)
    public ResSections: Array<ResSection> = new Array<ResSection>();
    @Type(()=>ResUser)
    public ResUsers: Array<ResUser> = new Array<ResUser>();
    @Type(()=>ResSurgeryRoom)
    public ResSurgeryRooms: Array<ResSurgeryRoom> = new Array<ResSurgeryRoom>();
    @Type(()=>ResSurgeryDesk)
    public ResSurgeryDesks: Array<ResSurgeryDesk> = new Array<ResSurgeryDesk>();
}
